main/glossary/README.md

```diff
@@ -33,19 +33,27 @@ disappear from the total allocation.
 
 ## AmountMath
 The AmountMath library executes the logic of how [amounts](#amounts) are changed when digital assets are merged, separated,
-or otherwise manipulated. For example, a deposit of 2 bucks into a purse that already has 3 bucks
-gives a new balance of 5 bucks. But, a deposit of a non-fungible theater ticket into a purse that already holds
-five tickets isn't done by numeric addition. Instead, you have to combine two arrays, containing either
-strings or objects/records.
-
-`AmountMath` has a single set of polymorphic
-methods of two different asset kinds to deal with [fungible](#fungible) assets (values are natural numbers) and
-[non-fungible](#non-fungible) assets (values are an array or object). The two `AssetKinds` are
-- `AssetKind.NAT`: Used with fungible assets. Amount values are natural numbers (non-negative integers). Default value.
-- `AssetKind.SET`: Used with non-fungible assets. Amount values are
-  arrays of strings, numbers, objects, or other comparable values.
-  Values should never include promises (they aren't comparable), or
-  payments, purses, and anything else that can't be shared freely.
+or otherwise manipulated. For example, a deposit of 2 [Quatloos](#quatloos) into a purse that already has 3 Quatloos
+gives a new balance of 5 Quatloos. But a deposit of a non-fungible theater ticket into a purse that already holds
+five tickets is performed by set union rather than by arithmetic.
+
+`AmountMath` has a single set of polymorphic methods that deal with different asset kinds:
+- `AssetKind.NAT`: Used with [fungible](#fungible) assets.
+  Each amount value is a natural number (non-negative integer).
+  This is the default `AssetKind`.
+- `AssetKind.SET`: Used with [non-fungible](#non-fungible) assets;
+  deprecated in favor of `AssetKind.COPY_SET` but still in wide use.
+  Each amount value is an array of [key](#key) values
+  subject to the same constraints as those of `AssetKind.COPY_SET`.
+- `AssetKind.COPY_SET`: Used with [non-fungible](#non-fungible) assets.
+  Each amount value is a set of [key](#key) values
+  (strings, numbers, objects, etc.).
+  Values cannot include promises (they aren't keys), and should not
+  include privileged objects such as payments and purses.
+- `AssetKind.COPY_BAG`: Used with semi-fungible assets.
+  Each amount value is a [multiset](https://en.wikipedia.org/wiki/Multiset)
+  of [key](#key) values subject to the same constraints as
+  those of `AssetKind.COPY_SET` but allowed to be present more than once.
```

Comment:
@erights I've added the text for COPY_SET and COPY_BAG. Please (re-)review, and if possible suggest some text for "semi-fungible" (which is currently absent, a glaring omission here where we link to "fungible" and "non-fungible").

Reply:
## Semi-fungible
A semi-fungible asset is one where there are distinct forms of the asset, which are not interchangeable with each other, but in which there may be multiple instances of each form which are interchangeable with other instances of the same form. As a variation on the [non-fungible](#non-fungible) theater ticket example, perhaps the tickets only specify a section, where the holder may sit in any seat of that section. For each section, the number of theater tickets minted should not exceed the number of seats in that section.

Reply:
Waddaya think?